File: generated_code/phi4/baseline_code_java/code_row_58.java
import java.util.HashMap;
import java.util.Scanner;

class LoginSystem {
    private static HashMap<String, String> users = new HashMap<>();
    private static String loggedInUserEmail = null;
    
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        // Initialize with a dummy user for testing
        users.put("user@example.com", "password123");

        while (true) {
            System.out.println("\n1. Login");
            System.out.println("2. Change Email");
            System.out.println("3. Exit");

            int choice = scanner.nextInt();
            scanner.nextLine(); // Consume newline

            switch (choice) {
                case 1:
                    login(scanner);
                    break;
                case 2:
                    changeEmail(scanner);
                    break;
                case 3:
                    System.exit(0);
                    break;
                default:
                    System.out.println("Invalid choice. Please try again.");
            }
        }
    }

    private static void login(Scanner scanner) {
        if (loggedInUserEmail != null) {
            System.out.println("You are already logged in.");
            return;
        }

        System.out.print("Enter email: ");
        String email = scanner.nextLine();

        System.out.print("Enter password: ");
        String password = scanner.nextLine();

        if (users.containsKey(email) && users.get(email).equals(password)) {
            loggedInUserEmail = email;
            System.out.println("Login successful.");
        } else {
            System.out.println("Invalid credentials. Please try again.");
        }
    }

    private static void changeEmail(Scanner scanner) {
        if (loggedInUserEmail == null) {
            System.out.println("You must be logged in to change your email.");
            return;
        }

        System.out.print("Enter old email: ");
        String oldEmail = scanner.nextLine();

        System.out.print("Confirm password: ");
        String confirmPassword = scanner.nextLine();

        if (!oldEmail.equals(loggedInUserEmail)) {
            System.out.println("Old email does not match the logged-in user's email.");
            return;
        }

        if (!users.get(oldEmail).equals(confirmPassword)) {
            System.out.println("Incorrect password. Email change failed.");
            return;
        }

        System.out.print("Enter new email: ");
        String newEmail = scanner.nextLine();

        if (users.containsKey(newEmail)) {
            System.out.println("New email is already in use.");
            return;
        }

        users.remove(oldEmail);
        users.put(newEmail, confirmPassword);

        loggedInUserEmail = newEmail;

        System.out.println("Email changed successfully.");
    }
}

File: generated_code/qwen/baseline_code_py/code_row_58.py
from getpass import getpass

class User:
    def __init__(self, username, email, password):
        self.username = username
        self.email = email
        self.password = password

class LoginSystem:
    def __init__(self):
        self.users = {}
    
    def register_user(self, username, email, password):
        if username in self.users:
            print("Username already exists.")
            return False
        self.users[username] = User(username, email, password)
        print("User registered successfully.")
        return True
    
    def login(self, username, password):
        user = self.users.get(username)
        if user and user.password == password:
            print("Login successful.")
            return user
        else:
            print("Invalid username or password.")
            return None

    def change_email(self, logged_in_user, old_email, new_email):
        if logged_in_user.email != old_email:
            print("Old email does not match the current email.")
            return False
        logged_in_user.email = new_email
        print("Email changed successfully.")
        return True

def main():
    system = LoginSystem()
    
    # Register a user for demonstration purposes
    system.register_user("testuser", "oldemail@example.com", "password123")
    
    username = input("Enter your username: ")
    password = getpass("Enter your password: ")
    
    logged_in_user = system.login(username, password)
    
    if logged_in_user:
        old_email = input("Enter your old email: ")
        new_email = input("Enter your new email: ")
        
        system.change_email(logged_in_user, old_email, new_email)

if __name__ == "__main__":
    main()